tools/src/main/scala/org/apache/spark/sql/rapids/tool/ui/QualificationReportGenerator.scala

```diff
@@ -0,0 +1,168 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.rapids.tool.ui
+
+import java.nio.file
+import java.nio.file.{Files, FileSystems, Paths}
+
+import scala.collection.JavaConverters.mapAsJavaMapConverter
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileSystem, FSDataOutputStream, Path}
+import org.json4s.DefaultFormats
+import org.json4s.jackson.Serialization
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.rapids.tool.qualification.QualificationSummaryInfo
+import org.apache.spark.util.Utils
+
+class QualificationReportGenerator(outputDir: String,
+    sumArr: Seq[QualificationSummaryInfo]) extends Logging {
+
+  import QualificationReportGenerator._
+  implicit val formats = DefaultFormats
+
+  val outputWorkPath = new Path(outputDir)
+  val fs = Some(FileSystem.get(outputWorkPath.toUri, new Configuration()))
+  val sums = Some(sumArr)
+
+  def launch(): Unit = {
+    val uiRootPath = getPathForResource(RAPIDS_UI_ASSETS_DIR)
+    logInfo(s"Generating UI files into... ${outputWorkPath.toUri}")
+    copyAssetFolderRecursively(uiRootPath, outputWorkPath)
+  }
+
+  def copyAssetFolderRecursively(srcFolderPath: java.nio.file.Path, dstPath: Path): Unit = {
+    logInfo(s"UI code generator: Copying ... ${srcFolderPath.toUri}")
+    if (Files.isDirectory(srcFolderPath)) {
+      val destinationPath = new Path(dstPath, srcFolderPath.getFileName.toString)
+      fs.map { dstFileSys =>
+        dstFileSys.mkdirs(destinationPath)
+        Files.list(srcFolderPath).forEach { childPath =>
+          if (Files.isDirectory(childPath)) {
+            copyAssetFolderRecursively(childPath, destinationPath)
+          } else {
+            tryCopyAssetFile(childPath, new Path(destinationPath, childPath.getFileName.toString))
+          }
+        }
+      }
+    }
+  }
+
+  def generateJSFiles(): Unit = {
+    // Serializing the entire list of sums may stress the memory.
+    // Serializing one record at a time is slower but it would reduce the memory peak consumption.
+    val outputPath = new Path(outputWorkPath, RAPIDS_UI_JS_DATA)
+    logInfo(s"Generating UI data in ${outputPath.toUri.toString}")
+    val fileHeader =
+      s"""
+        |let qualificationRecords = [
+       """.stripMargin
+    val fileFooter =
+      s"""|];
+       """.stripMargin
+    fs.foreach { dfs =>
+      val outputPutFile = Some(dfs.create(outputPath))
```

Comment:
doesn't seem like you need this as Option just use  it and remove the outputPutFile.foreach